docs/configuration/config-file-reference.md

```diff
@@ -2673,12 +2673,19 @@ The `limits_config` configures default and per-tenant limits imposed by Cortex s
 # CLI flag: -querier.max-fetched-series-per-query
 [max_fetched_series_per_query: <int> | default = 0]
 
-# The maximum size of all chunks in bytes that a query can fetch from each
-# ingester and storage. This limit is enforced in the querier and ruler only
-# when running Cortex with blocks storage. 0 to disable.
+# Deprecated (user max-fetched-data-bytes-per-query instead): The maximum size
+# of all chunks in bytes that a query can fetch from each ingester and storage.
+# This limit is enforced in the querier and ruler only when running Cortex with
+# blocks storage. 0 to disable.
 # CLI flag: -querier.max-fetched-chunk-bytes-per-query
 [max_fetched_chunk_bytes_per_query: <int> | default = 0]
 
+# The maximum combined size of all data that a query can fetch from each
+# ingester and storage. This limit is enforced in the querier and ruler only
+# when running Cortex with blocks storage. 0 to disable.
+# CLI flag: -querier.max-fetched-data-bytes-per-query
+[max_fetched_data_bytes_per_query: <int> | default = 0]
```

Comment:
Does this limit apply only to `Select` call? Since I don't see we are limiting label names and label values. If that's the case probably we can mention it in the doc or give the flag a better name.

Reply:
That's the only question I have now. Other LGTM

Reply:
The naming of the limit is consistent with other similar limits. I think we can implement this limit also on the LabelNames and LabelValues as well. Right now I've updated the doc saying that it is only applied for query, query_range and series APIs

Reply:
NIT: `when running Cortex with blocks storage`

Currently we only have block storage as chunk storage was removed.

Reply:
All the other limits have this line:
```
This limit is enforced in the querier only when running Cortex with blocks storage
```
Should we remove it everywhere?

Reply:
I think we should...

Reply:
I'll also address  this in the PR to handle retryable failures.